### PR TITLE
Dockerize tests

### PR DESCRIPTION
**Pull request by fredsmith**
This moves travis tests into docker, running on 16.04. 

You can run: `make travis_up` to pull up the complete runtime environment, and `make travis_test` to run unit tests, and then `make travis_down` to tear down the running instances

### CODE REVIEW

File: .travis/run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -xe
+. /edx/app/edx_notes_api/venvs/edx_notes_api/bin/activate
+
+cd /edx/app/edx_notes_api/edx_notes_api
+
+
+export ELASTICSEARCH_URL=http://es:9200/
```
Comment (feanil) [inline .travis/run_tests.sh]: We don't need this, we get them from the compose file.

**Comment by fredsmith** [comment]
@muzaffaryousaf Hey, this test (test_search_course) is failing when I try to build this PR.  It seems to have ended up searching case-insensitive, when it was designed to do a case sensitive search.  I don't think I changed anything that should have affected this - I've tried with both elasticsearch 0.9.11 and 1.3.0

File: Makefile [hunk 2]
```diff
@@ -1,16 +1,12 @@
...
 .PHONY: requirements
 
-validate: test.requirements test coverage
+include .travis/docker.mk
 
-ifeq ($(ESVER),-)
-test_settings = notesserver.settings.test_es_disabled
-else
-test_settings = notesserver.settings.test
-endif
```
Comment (muzaffaryousaf) [inline Makefile]: @fredsmith I think different setting could be the reason for failure, not sure why we removed these settings.

Comment (muzaffaryousaf) [inline Makefile]: NVM that's not the reason. ignore.

**Comment by muzaffaryousaf** [comment]
@fredsmith Any idea how i can run these tests locally, same as before ?

**Comment by fredsmith** [comment]
The test settings file now takes configuration via environment settings.  The defaults assume SQLite and elastic search on local host.  Then make validate, same as before.   You can run the Travis tests locally with `make travis_up; make travis_test; make travis_down`

File: .travis/docker-compose-travis.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    container_name: db
+    command: mysqld --character-set-server=utf8 --collation-server=utf8_general_ci
+    environment:
+      MYSQL_ROOT_PASSWORD: ""
+      MYSQL_ALLOW_EMPTY_PASSWORD: "yes"
+      MYSQL_USER: "notes001"
+      MYSQL_PASSWORD: "secret"
+      MYSQL_DATABASE: "edx_notes_api"
+  es:
+    image: edxops/elasticsearch:0.9.13
```
Comment (muzaffaryousaf) [inline .travis/docker-compose-travis.yml]: @fredsmith  can you try with `1.4.2` too.

Comment (muzaffaryousaf) [inline .travis/docker-compose-travis.yml]: Ignore

**Comment by muzaffaryousaf** [comment]
@fredsmith Earlier we were using sqlite(case sensitive) db for tests but for this PR we started to use mysql which is case insensitive by default that's why the test was failing. I think this test does't make sense since we don't explicitly make search case sensitive for course_id. I'd prefer to check case insensitivity for text parameter where we make it case insensitive explicitly.

**Comment by muzaffaryousaf** [comment]
@fredsmith Build is green and i hope you're unblock on this now. Let me know if i can help. Thanks

**Comment by fredsmith** [comment]
@muzaffaryousaf was that test searching elasticsearch or SQL?

**Pull request merged**